hudi-client/src/test/java/org/apache/hudi/table/action/rollback/TestCopyOnWriteRollbackActionExecutor.java

```diff
@@ -0,0 +1,246 @@
+    HoodieWriteConfig config = HoodieWriteConfig.newBuilder().withPath(basePath)
+        .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.INMEMORY).build()).build();
+    HoodieTable table = this.getHoodieTable(metaClient, config);
+    HoodieInstant needRollBackInstant = new HoodieInstant(false, HoodieTimeline.COMMIT_ACTION, "002");
+
+    // execute CopyOnWriteRollbackActionExecutor with filelisting mode
+    CopyOnWriteRollbackActionExecutor copyOnWriteRollbackActionExecutor = new CopyOnWriteRollbackActionExecutor(jsc, config, table, "003", needRollBackInstant, true);
+    assertFalse(copyOnWriteRollbackActionExecutor.getRollbackStrategy() instanceof MarkerBasedRollbackStrategy);
+    List<HoodieRollbackStat> hoodieRollbackStats = copyOnWriteRollbackActionExecutor.executeRollback();
+
+    // assert hoodieRollbackStats
+    assertEquals(hoodieRollbackStats.size(), 3);
+    hoodieRollbackStats.forEach(stat -> {
+      if (stat.getPartitionPath().equals("2015/03/16")) {
+        assertEquals(1, stat.getSuccessDeleteFiles().size());
+        assertEquals(0, stat.getFailedDeleteFiles().size());
+        assertEquals(null, stat.getCommandBlocksCount());
+        assertEquals("file:" + HoodieTestUtils.getDataFilePath(basePath, "2015/03/16", commitTime2, file21),
+            stat.getSuccessDeleteFiles().get(0));
+      } else if (stat.getPartitionPath().equals("2015/03/17")) {
+        assertEquals(1, stat.getSuccessDeleteFiles().size());
+        assertEquals(0, stat.getFailedDeleteFiles().size());
+        assertEquals(null, stat.getCommandBlocksCount());
+        assertEquals("file:" + HoodieTestUtils.getDataFilePath(basePath, "2015/03/17", commitTime2, file22),
+            stat.getSuccessDeleteFiles().get(0));
+      } else if (stat.getPartitionPath().equals("2015/03/17")) {
+        assertEquals(0, stat.getSuccessDeleteFiles().size());
+        assertEquals(0, stat.getFailedDeleteFiles().size());
+        assertEquals(null, stat.getCommandBlocksCount());
+      }
+    });
+
+    assertTrue(HoodieTestUtils.doesCommitExist(basePath, "001"));
+    assertTrue(HoodieTestUtils.doesInflightExist(basePath, "001"));
+    assertFalse(HoodieTestUtils.doesCommitExist(basePath, "002"));
+    assertFalse(HoodieTestUtils.doesInflightExist(basePath, "002"));
+    assertTrue(HoodieTestUtils.doesDataFileExist(basePath, "2015/03/16", commitTime1, file11)
+        && HoodieTestUtils.doesDataFileExist(basePath, "2015/03/17", commitTime1, file12));
+    assertFalse(HoodieTestUtils.doesDataFileExist(basePath, "2015/03/16", commitTime2, file21)
+        || HoodieTestUtils.doesDataFileExist(basePath, "2015/03/17", commitTime2, file22));
+  }
+
+  private void twoUpsertCommitDataRollBack(boolean isUsingMarkers) throws IOException {
+    //1. prepare data
+    HoodieWriteConfig cfg = getConfigBuilder().withRollbackUsingMarkers(isUsingMarkers).build();
```

Comment:
cc @xushiyan I feel just making these two commits or this template code in some TestHelper.. will clean up code by a lot..